elements/src/components/pos-list/pos-list.integration.spec.tsx

```diff
@@ -0,0 +1,65 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { mockPodOS } from '../../test/mockPodOS';
+import { PosApp } from '../pos-app/pos-app';
+import { PosLabel } from '../pos-label/pos-label';
+import { PosList } from './pos-list';
+import { PosResource } from '../pos-resource/pos-resource';
+import { when } from 'jest-when';
+
+describe('pos-list', () => {
+  it('children render label for loaded resources (without fetching)', async () => {
+    const os = mockPodOS();
+    when(os.store.get)
+      .calledWith('https://resource.test')
+      .mockReturnValue({
+        relations: () => [
+          {
+            predicate: 'http://schema.org/video',
+            uris: ['https://video.test/video-1', 'https://video.test/video-2'],
+          },
+        ],
+      });
+    when(os.store.get)
+      .calledWith('https://video.test/video-1')
+      .mockReturnValue({ uri: 'https://video.test/video-1', label: () => 'Video 1' });
+    when(os.store.get)
+      .calledWith('https://video.test/video-2')
+      .mockReturnValue({ uri: 'https://video.test/video-2', label: () => 'Video 2' });
+    const page = await newSpecPage({
+      components: [PosApp, PosLabel, PosList, PosResource],
+      supportsShadowDom: false,
+      html: `
+      <pos-app>
+        <pos-resource uri="https://resource.test" lazy="">
+          <pos-list rel="http://schema.org/video">
+            <template>
+              <pos-label />
+            </template>
+          </pos-list>
+        </pos-resource>
+      </pos-app>`,
+    });
+    expect(os.fetch.mock.calls).toHaveLength(0);
+    expect(page.root?.querySelectorAll('pos-list pos-resource')).toHaveLength(2);
+    const label1 = page.root?.querySelectorAll('pos-list pos-resource')[0] as unknown as PosResource;
```

Comment:
Let's not repeat the `querySelectorAll` here, but extract the result in a variable